From b7b651c8db5063b7ed6de08595a3c456573610b0 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Tue, 18 Jun 2019 12:10:55 +0530
Subject: [PATCH] compilers: Fix bitcode and other options for objc code

We were setting the base options for the Objective-C compiler
manually, due to which options such as b_bitcode and b_ndebug were not
getting set at all.

The base options here are the same as for C code with the Clang
compiler, so just use the same inherited list.
---
 mesonbuild/compilers/objc.py   | 1 -
 mesonbuild/compilers/objcpp.py | 1 -
 2 files changed, 2 deletions(-)

diff --git a/mesonbuild/compilers/objc.py b/mesonbuild/compilers/objc.py
index 5b2b517f..7f3431c9 100644
--- a/mesonbuild/compilers/objc.py
+++ b/mesonbuild/compilers/objc.py
@@ -68,4 +68,3 @@ class ClangObjCCompiler(ClangCompiler, ObjCCompiler):
         self.warn_args = {'1': default_warn_args,
                           '2': default_warn_args + ['-Wextra'],
                           '3': default_warn_args + ['-Wextra', '-Wpedantic']}
-        self.base_options = ['b_pch', 'b_lto', 'b_pgo', 'b_sanitize', 'b_coverage']
diff --git a/mesonbuild/compilers/objcpp.py b/mesonbuild/compilers/objcpp.py
index e1b7a7dc..4a8aa51d 100644
--- a/mesonbuild/compilers/objcpp.py
+++ b/mesonbuild/compilers/objcpp.py
@@ -69,4 +69,3 @@ class ClangObjCPPCompiler(ClangCompiler, ObjCPPCompiler):
         self.warn_args = {'1': default_warn_args,
                           '2': default_warn_args + ['-Wextra'],
                           '3': default_warn_args + ['-Wextra', '-Wpedantic']}
-        self.base_options = ['b_pch', 'b_lto', 'b_pgo', 'b_sanitize', 'b_coverage']
-- 
2.21.0

